src/ripple/app/ledger/impl/InboundLedgers.cpp

```diff
@@ -200,14 +201,19 @@ class InboundLedgersImp
         return ret;
     }
 
-    void logFailure (uint256 const& h)
+    void logFailure (uint256 const& h, std::uint32_t seq)
     {
-        mRecentFailures.insert (h);
+        ScopedLockType sl (mLock);
+
+        mRecentFailures.emplace(h, seq);
```

Comment:
Hmm, so it isn't really "logging" anything here - everything's still in memory.  why not call this `recordFailure`?